reconcile/gitlab_permissions.py

```diff
@@ -26,6 +27,25 @@ def get_members_to_add(repo, gl, app_sre):
     return members_to_add
 
 
+def get_all_app_sre_maintainers(repo, gl, app_sre):
+    app_sre_user_ids = [user.id for user in app_sre]
+    app_sre_maintainers = []
+    chunks = [
+        app_sre_user_ids[i : i + PAGE_SIZE]
+        for i in range(0, len(app_sre_user_ids), PAGE_SIZE)
+    ]
+    for chunk in chunks:
+        maintainers = gl.get_project_maintainers(
+            repo, query=create_user_ids_query(chunk)
+        )
+        app_sre_maintainers.extend(maintainers)
+    return app_sre_maintainers
```

Comment:
in python 3.12, there is [itertools.batched](https://docs.python.org/3.12/library/itertools.html#itertools.batched) can do chunk, can add it to util and doc can be replaced in 3.12


```python
def batched(iterable, n):
    # batched('ABCDEFG', 3) → ABC DEF G
    if n < 1:
        raise ValueError('n must be at least one')
    it = iter(iterable)
    while batch := tuple(islice(it, n)):
        yield batch
```

Then this logic can be simplified like

```python
    app_sre_user_ids = [user.id for user in app_sre]
    chunks = batched(app_sre_user_ids, PAGE_SIZE)
    app_sre_maintainers = (
        gl.get_project_maintainers(repo, query=create_user_ids_query(chunk))
        for chunk in chunks
    )
    return list(itertools.chain.from_iterable(app_sre_maintainers))
```

Reply:
@hemlso updated the code